### PR TITLE
Prefix recovery tokens for deeplinking to native apps

### PR DESCRIPTION
**Pull request by coldlink**
## What does this change?

The reason we do this is that this allows apps to distinguish between each other's deeplinks, for example between the android live app and the android puzzles app (which will have a different prefix) e.g `/reset-password/al_token` and `/reset-password/ap_token` and also allows us to have multiple paths under the same path which should not be intercepted by the app e.g. `/reset-password/email-sent`.

The apps will have to intercept links based on the path + prefix for example, the android live app uses the prefix 'al_'
therefore the deeplink path for reset password would be: `/reset-password/al_<token>`.
Where <token> is the okta recovery token, and the android live app will intercept the deeplink on the `/reset-password/al_*` path.

So this PR adds two methods, one for extracting the recovery token from a parameter which may or may not have a prefix, and one for adding the prefix onto a recovery token should the `appClientId` parameter match.

This will then be sent to the user via email, the behaviour of the tokens is the same regardless of if they have the prefix or not. However the apps can intercept any link which has a prefix they need.

## Other changes

- Updates the `PasswordForm` (used on the `Welcome` page) to add a spinner to the `Save and continue` button

### CODE REVIEW

File: src/server/lib/deeplink/oktaRecoveryToken.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ * @param token string representing the recovery token
+ * @param appClientId string representing the app client id
+ * @returns string representing the recovery token with the app prefix
+ *
+ */
+export const addAppPrefixToOktaRecoveryToken = async (
+  token: string,
+  appClientId?: string,
+  request_id?: string,
```
Comment (raphaelkabo) [inline src/server/lib/deeplink/oktaRecoveryToken.ts:63]: Could we do away with `request_id`? We only use it for error logging, right? Is there a particular benefit to keeping it in the logs? I ask mostly because otherwise it feels kind of irrelevant to this function.

File: src/server/lib/deeplink/oktaRecoveryToken.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import { getApp } from '@/server/lib/okta/api/apps';
```
[the remark this answers is not in the document]
Comment (coldlink) [reply src/server/lib/deeplink/oktaRecoveryToken.ts:1]: I felt like just using `deeplink` didn't adequately describe the usage, while `deeplink/oktaRecoveryToken` did. Which was the main reasoning.

[the remark this answers is not in the document]
Comment (coldlink) [reply src/server/lib/deeplink/oktaRecoveryToken.ts:1]: This way you know it's something related to deeplinking, and also related to the recovery token, rather than one or the other, without having to go into the file.

File: src/server/lib/deeplink/oktaRecoveryToken.ts (hunk 1 above)
Comment (coldlink) [reply src/server/lib/deeplink/oktaRecoveryToken.ts:63]: Yeah, but it's super useful for error logging, hence it's passed around everywhere. There isn't another way to pass around this id, as we don't have any way of injecting additional metadata per request.

Comment (raphaelkabo) [reply src/server/lib/deeplink/oktaRecoveryToken.ts:63]: If we use it for debugging, we should keep it!

File: src/server/lib/deeplink/oktaRecoveryToken.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (raphaelkabo) [reply src/server/lib/deeplink/oktaRecoveryToken.ts:1]: This seems reasonable.

**Review by raphaelkabo — APPROVED** [review]
This looks really solid to me, no notes beyond the ones which have already been addressed. You may want someone else to look through it, but since it's been extensively tested with apps, I think it's good to go.
